pkg/reconciler/configuration/resources/revision.go

```diff
@@ -40,8 +40,17 @@ func MakeRevision(config *v1alpha1.Configuration) *v1alpha1.Revision {
 
 	UpdateRevisionLabels(rev, config)
 
-	if rev.Annotations == nil {
-		rev.Annotations = make(map[string]string)
+	// Populate the CreatorAnnotation from configuration.
+	cans := config.GetAnnotations()
+	creator, ok := cans[serving.CreatorAnnotation]
+	if ok {
+		rans := rev.GetAnnotations()
+		if rans == nil {
+			rans = map[string]string{serving.CreatorAnnotation: creator}
+			defer rev.SetAnnotations(rans)
```

Comment:
You shouldn't need the `defer` here. Also It might be more clear to do the following so you only have one way to set creator instead of having the else block.

```
...
if rans == nil {
  rans = map[string]string{}
  rev.SetAnnotations(rans)
}

rans[serving.CreatorAnnotation] = creator
```